sox.features.js

```diff
@@ -2620,6 +2620,26 @@
       const answers = document.querySelector("#answers-header h2").innerText.split(' ')[0];
       const toInsert = '<div class="grid--cell ws-nowrap mb8 ml16"><span class="fc-light mr4">Answers</span> ' + answers + '</div>';
       document.querySelector(".d-flex.fw-wrap").insertAdjacentHTML("beforeEnd", toInsert);
-    }
+    },
+
+    customizeToolsPageLists: function(settings) {
+        const trimLists = function(table) {
+            if (settings.filterInvalid) {
+                table.querySelectorAll("td.tagged-ignored").forEach(cell => cell.parentNode.remove());
+            }
+            if (settings.filterTypeQuestions) {
+                table.querySelectorAll("a.question-hyperlink").forEach(cell => cell.parentNode.parentNode.remove());
+            }
+            if (settings.filterTypeAnswers) {
+                table.querySelectorAll("a.answer-hyperlink").forEach(cell => cell.parentNode.parentNode.remove());
+            }
+            const listCount = settings.listCount || 3;
+            const rows = table.querySelectorAll("tr");
+            rows.forEach((row, i) => row.classList.toggle("collapsing", i >= listCount));
+        };
+        // tables are populated by XHR after page loads
+        sox.helpers.observe(Array.from(document.querySelectorAll("div.island")), "table.summary-table", trimLists);
```

Comment:
We could replace this with `sox.helpers.addAjaxListener` like this:

```
      // tables are populated by XHR after page loads
      sox.helpers.addAjaxListener("\\/tools\\?tab=", () => {
        document.querySelectorAll("table.summary-table").forEach(trimLists);
      });
      // just in case we come in after a table has been loaded already
      document.querySelectorAll("table.summary-table").forEach(trimLists);
```

we found that the advantage of doing it this way is we stop adding so many mutation observers to the page, which seemed to have negative performance effects when there were too many. 

This method instead just triggers your function whenever an xhr response from the given URL is received, after which we can query for the table once